lib/spack/spack/environment/environment.py

```diff
@@ -825,6 +825,79 @@ def _process_definition(self, item):
             else:
                 self.spec_lists[name] = user_specs
 
+    def _process_views(self, env_view):
+        """Process view option(s), which can be boolean, string, or None.
+
+        A boolean environment view option takes precedence over any that may
+        be included. So ``view: True`` results in the default view only. And
+        ``view: False`` means the environment will have no view.
+
+        Any other value means the FIRST include view option will be combined
+        with that of the environment (if any) to establish the view(s). If
+        there is still no ``view`` then the default view will be used.
+
+        Args:
+            env_view: view option provided in the manifest or ``None``.
+        """
+
+        def add_view(name, values):
+            if name not in self.views:
+                if isinstance(values, str):
+                    self.views[name] = ViewDescriptor(self.path, values)
+                elif isinstance(values, dict):
+                    self.views[name] = ViewDescriptor.from_dict(self.path, values)
+                else:
+                    tty.error(f"Cannot add view named {name} for {type(values)} values {values}")
+            else:
+                tty.debug(f"Already have '{name}' view ({self.views[name]}), skipping {values}")
+
+        self.views = {}
+
+        # If the environment specifies 'view: False' then the environment is
+        # to have NO views regardless of whether included views are specified.
+        if env_view is False:
+            return
+
+        # If the environment specifies 'view: True' then only the default
+        # view will be created for the environment.
+        if env_view is True:
+            add_view(default_view_name, self.view_path_default)
+            return
+
+        # Otherwise, we have a subdirectory or dictionary
+        if isinstance(env_view, str):
+            add_view(default_view_name, env_view)
+        elif env_view:
+            for name, values in env_view.items():
+                add_view(name, values)
```

Comment:
This all appears to be redundant (as is the argument to this method).

The include order test shows that if you have `view: true` at the highest level (in this case in the environment) will override the entire dictionary for lower-priority scopes. If you remove this block you can then also remove the TBD comments below, as we guarantee that the view priority follows from the same YAML merging semantics.

Reply:
The logic is similar for processing the view from the environment and the included views since they support the meaning of each type argument.  But the environment configuration from the manifest file -- passed in to `_process_views` as `env_view` is not the same as the included views we get from `spack config`.

Reply:
Also, which test is the `include order test`?

Reply:
`test_env_include_mixed_views` is the one that covers the include ordering.

Why wouldn't be the config from the config be the same as the config from the file? The solution should be to fix that so they're the same (I don't see anything in the code to suggest they wouldn't be, but would need to check that) not to overcomplicate the code.

Reply:
@becker33 I refactored view processing to hopefully be simpler (after removing the redundant view from the schema).  However, the spack configuration scopes still do not reflect the manifest contents.  As a result, as is done with included definitions, it is currently necessary to process the manifest's view separate from the configuration in order to get all of the desired behavior as represented by the tests.

I also kept the `add_view` as it is still useful for processing view dictionary contents (that can contain strings (for paths) or dictionaries (for named views)). 

I am still skipping the inclusion of a named view if the view name already exists since there is potential for conflicting views having the same name but can revisit if you think it's warranted.